source/fundamentals/data-formats/document-data-format-pojo.txt

```diff
@@ -13,17 +13,227 @@ Document Data Format: POJOs
+In the following sections we show:
 
-- Brief introduction, link to data-formats/codecs for more info
-- Example demonstrating the following:
-  - specifying how to create a CodecRegistry that uses PojoCodecProvider
-  - example POJO class with getters/setters
-  - include user-defined classes and lists
-- Sample input/output of aforementioned example
+- a POJO example
+- how to configure the driver to serialize and deserialize the POJO
+- sample code to read and write documents to MongoDB using the POJO
 
-- Link to data-formats/pojo-customization for more information
+.. _fundamentals-example-pojo:
+
+Example POJO
+------------
+
+To follow the steps in this guide, you can create your own class or use the
+following sample POJO class which describes characteristics of a flower:
+
+.. code-block:: java
+
+   public class Flower {
+
+       private ObjectId id;
+       private String name;
+       private List<String> colors;
+       private Boolean isBlooming;
+       private Float height;
+
+       // public empty constructor needed for retrieving the POJO
+       public Flower() {}
+
+       public Flower(String name, Boolean isBlooming, Float height, List<String> colors) {
+           this.name = name;
+           this.isBlooming = isBlooming;
+           this.height = height;
+           this.colors = colors;
+       }
+
+       public ObjectId getId() {
+           return id;
+       }
+
+       public void setId(ObjectId id) {
+           this.id = id;
+       }
+
+       public String getName() {
+           return name;
+       }
+
+       public void setName(String name) {
+           this.name = name;
+       }
+
+       public Boolean getIsBlooming() {
+           return isBlooming;
+       }
+
+       public void setIsBlooming(Boolean isBlooming) {
+           this.isBlooming = isBlooming;
+       }
+
+       public Float getHeight() {
+           return height;
+       }
+
+       public void setHeight(Float height) {
+           this.height = height;
+       }
+
+       public List<String> getColors() {
+           return colors;
+       }
+
+       public void setColors(List<String> colors) {
+           this.colors = colors;
+       }
+
+       @Override
+       public String toString() {
+           return "Flower [id=" + id + ", name=" + name + ", colors=" + colors + ", isBlooming=" + isBlooming + ", height=" + height + "]";
+       }
+   }
+
+If you are creating your own POJO for storing and retrieving data in MongoDB,
+make sure to follow the following guidelines:
+
+- The POJO class should not implement interfaces or extend classes from a
+  framework.
+- Include all the fields for which you want to store and retrieve data;
+  make sure they are not ``static`` or ``transient``.
+- If you include public getter or setter methods using the
+  `JavaBean naming conventions <https://docs.oracle.com/javase/tutorial/javabeans/writing/properties.html>`__
+  in your POJO, they get called when serializing or deserializing data.  If
```

Comment:
nit: "the get called" sounds informal compared to the rest of this document, and could be confusing since it's close to multiple occurrences of the word "getters". I'd suggest rewording to something like "the driver calls those methods when serializing or deserializing data".

Reply:
Thanks for spotting this, and good suggestion.